### PR TITLE
[Enhancement] (nereids)implement showCollationCommand in nereids

### LINKED ISSUE #42748: [Enhancement] (nereids)implement showCollationCommand in nereids
showCollation command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showCollation command in nereids:
unsupportedshowCollationStatement
``` SHOW COLLATION wildWhere?                                                     ```

### PR DESCRIPTION
**Pull request by msridhar78**
Issue Number: close #42748 

implement showCollationCommand in nereids

mysql> show collation;
+--------------------+---------+------+---------+----------+---------+
| Collation          | Charset | Id   | Default | Compiled | Sortlen |
+--------------------+---------+------+---------+----------+---------+
| utf8mb4_0900_bin   | utf8mb4 | 309  | Yes     | Yes      | 1       |
| utf8mb3_general_ci | utf8mb3 | 33   | Yes     | Yes      | 1       |
+--------------------+---------+------+---------+----------+---------+
2 rows in set (22.40 sec)

### CODE REVIEW

**Comment by msridhar78** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCollationCommand.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.doris.qe.StmtExecutor;
+
+import com.google.common.collect.Lists;
+
+import java.util.List;
+
+/**
+ * Represents the command for SHOW COLLATION
+ */
+public class ShowCollationCommand extends Command implements NoForward {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCollationCommand.java:37]:
```suggestion
public class ShowCollationCommand extends ShowCommand {
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java [hunk 2]
```diff
@@ -3999,4 +4001,17 @@ public SetUserPropertiesCommand visitSetUserProperties(SetUserPropertiesContext
     public SetDefaultStorageVaultCommand visitSetDefaultStorageVault(SetDefaultStorageVaultContext ctx) {
         return new SetDefaultStorageVaultCommand(stripQuotes(ctx.identifier().getText()));
     }
+
+    @Override
+    public LogicalPlan visitShowCollation(ShowCollationContext ctx) {
+        String wild = null;
+        if (ctx.wildWhere() != null) {
+            if (ctx.wildWhere().LIKE() != null) {
+                wild = ctx.wildWhere().STRING_LITERAL().getText().replace("'", "").trim();
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:4010]:
```suggestion
                wild = stripQuotes(ctx.wildWhere().STRING_LITERAL().getText());
```

**Comment by starocean999** [comment]
please add test case

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCollationCommand.java (hunk 1 above)
Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCollationCommand.java]: done - pls review

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Review by LiBinfeng-01 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**